fedml_api/standalone/domain_generalization/datasets/ser.py

```diff
@@ -0,0 +1,392 @@
+            waveform = torch.mean(waveform, dim=0, keepdim=True)
+
+        # Resample if necessary
+        if sample_rate != self.sample_rate:
+            resampler = torchaudio.transforms.Resample(sample_rate, self.sample_rate)
+            waveform = resampler(waveform)
+
+        # Generate mel spectrogram
+        mel_spectrogram = self.mel_spectrogram(waveform)
+
+        # Convert to decibels
+        log_mel_spectrogram = self.amplitude_to_db(mel_spectrogram)
+
+        # Convert to 3-channel image format for compatibility with image-based backbones
+        # Stack the same spectrogram 3 times to create RGB-like channels
+        spectrogram_image = log_mel_spectrogram.repeat(3, 1, 1)
+
+        # Apply additional transforms if provided
+        if self.transform is not None:
+            # Convert to PIL Image for compatibility with image transforms
+            spec_min = spectrogram_image.min()
+            spec_max = spectrogram_image.max()
+            spectrogram_image = (
+                (spectrogram_image - spec_min) / (spec_max - spec_min) * 255.0
+            )
+            spectrogram_image = spectrogram_image.byte().numpy()
+            spectrogram_image = Image.fromarray(spectrogram_image.transpose(1, 2, 0))
+            spectrogram_image = self.transform(spectrogram_image)
+
+        if self.target_transform is not None:
+            target = self.target_transform(target)
+
+        return spectrogram_image, target
+
+    def __len__(self) -> int:
+        return len(self.samples)
+
+
+class FedLeaSER(FederatedDataset):
+    """
+    Federated Speech Emotion Recognition Dataset with domain generalization
+    """
+
+    NAME = "fl_ser"
+    SETTING = "domain_skew"
+    DOMAINS_LIST = ["crema-d", "ravdess", "emo-db"]
+    percent_dict = {"crema-d": 0.01, "ravdess": 0.01, "emo-db": 0.01}
+
+    # Number of emotion classes - unified 6 classes
+    # Neutral (0), Angry (1), Fear (2), Happy (3), Disgust (4), Sad (5)
+    N_CLASS = 6
+
+    # Transform for spectrograms
+    Nor_TRANSFORM = transforms.Compose(
+        [
+            transforms.Resize((128, 128)),  # Resize spectrogram
+            transforms.RandomCrop(128, padding=4),
+            transforms.RandomHorizontalFlip(),
+            transforms.ToTensor(),
+            transforms.Normalize((0.485, 0.456, 0.406), (0.229, 0.224, 0.225)),
+        ]
+    )
+
+    def get_data_loaders(self, selected_domain_list: List[str] = []):
+        """
+        Get data loaders for the selected domains, or all domains if none specified
+        """
+        using_list = (
+            self.DOMAINS_LIST
+            if len(selected_domain_list) == 0
+            else selected_domain_list
+        )
+
+        nor_transform = self.Nor_TRANSFORM
+
+        train_dataset_list: list[AudioDataset] = []
+        test_dataset_list: list[AudioDataset] = []
+
+        test_transform = transforms.Compose(
+            [
+                transforms.Resize((128, 128)),  # Resize spectrogram
```

Comment:
this is way too big, I suggest to change it to 64, 64
```suggestion
                transforms.Resize((64, 64)),  # Resize spectrogram
```